### DISCUSSION TITLE
Is the argument --builtin-extensions-dir fully supported?

### DISCUSSION THREAD

**Discussion by avsthiago** [coder/code-server#5073] [Q&A] [answered]
Hi,

I'm trying to set up two locations to store extensions, built-in and user specific ones. 

I'm using the following command to start code-server.
```
code-server --extensions-dir /home/user/.extensions --builtin-extensions-dir /opt/code-server/extensions 
```

I can see and use the extensions inside the `--extensions-dir`, the problem is with the built in ones. They all appear as disabled and I couldn't find a way to enable them. 

This is a comparison of what I see when I use the arg `--builtin-extensions-dir` with code-server and VS Code locally. VS Code has a button to enable the extension.

![Screenshot 2022-04-07 at 17 41 43](https://user-images.githubusercontent.com/13870565/162239293-85de961f-80f1-4818-a90f-141003f55569.png)
<img width="541" alt="Screenshot 2022-04-07 at 17 43 48" src="https://user-images.githubusercontent.com/13870565/162240490-659a0ce2-c05d-4ea4-a854-97a31923c1d6.png">

Does anybody know if I'm missing something or if the arg `--builtin-extensions-dir` is still under development?

**Comment by jsjoeio** [comment]
Interesting. I see `--builtin-extensions-dir` used in our tests but it's not printed when you use `--help`. It may not be passed through correctly. @code-asher any ideas?

**Reply by code-asher** [reply]
Hmm it looks to me like we are passing it through.  We will have to poke into it more.

**Comment by avsthiago** [comment, marked as answer]
FYI, the solution/workaround for me was to install the extensions inside the default built-in extensions folder `/usr/lib/code-server/lib/vscode/extensions/`. I used the following command.

```
code-server \ 
--install-extension hashicorp.terraform@2.20.1 \
--install-extension ms-python.python@2022.4.1 \
--extensions-dir /usr/lib/code-server/lib/vscode/extensions/
```

**Comment by smktpd** [comment]
It is quite unfortunate that this discussion was marked as one that has an answer: the workaround of installing extensions into "system" dir `/usr/lib/code-server/lib/vscode/extensions/` is that those extensions then get treated as system ones: they aren't listed as installed extensions, yet they are running and can't be disabled.

I am trying to prepare a docker image with code-server for a group of engineers and we would like to have some set of extensions to be-preinstalled by default, but since every engineer mounts their own directory to `${HOME}` the default location for user-installed extensions ~/.local/share/code-server/extensions/ thus gets overwritten by their mount and they won't see the pre-installed extensions.
Since I would like to make some extensions preinstalled, but let user disable them if they wish so - I can't use `/usr/lib/code-server/lib/vscode/extensions/` as install dir.

**Comment by smktpd** [comment]
I've found this ticket about vscode: https://github.com/Microsoft/vscode/issues/56614#issuecomment-697961001
Someone there shared this link to VSCode docs: https://code.visualstudio.com/docs/setup/enterprise#_set-up-vs-code-with-preinstalled-extensions
Does code-server support something like `./bootstrap/extensions/` (for linux)?

**Reply by code-asher** [reply]
We are not doing anything with extensions, just passing flags straight through to VS Code, so I am not sure.  If VS Code supports it in core, then it should work in code-server, but if it is something they add on top of VS Code, then nope.

**Reply by code-asher** [reply]
Ah just noticed it says Windows only, so it would definitely not work with Linux, either way.
